examples/hybrid/sphere/deformation_flow.jl

```diff
@@ -23,412 +25,306 @@ import TerminalLoggers
-    rd2 = r2(λ, ϕ)
-
-    d1 = min(1, (rd1 / R_t)^2 + (zd / Z_t)^2)
-    d2 = min(1, (rd2 / R_t)^2 + (zd / Z_t)^2)
-
-    q1 = 1 / 2.0 * (1 + cos(pi * d1)) + 1 / 2.0 * (1 + cos(pi * d2))
-    q2 = 0.9 - 0.8 * q1^2
-    q3 = 0.0
-    if d1 < 0.5 || d2 < 0.5
-        q3 = 1.0
-    else
-        q3 = 0.1
-    end
+    λ = coord.long
+    z = coord.z
 
-    if (z > z_c) && (ϕ > ϕ_c - rad2deg(1 / 8)) && (ϕ < ϕ_c + rad2deg(1 / 8))
-        q3 = 0.1
-    end
-    q4 = 1 - 3 / 10 * (q1 + q2 + q3)
+    k = 10 * R / τ
+    λp = λ - FT(360) * t / τ
 
-    ρq1 = ρ_ref(z) * q1
-    ρq2 = ρ_ref(z) * q2
-    ρq3 = ρ_ref(z) * q3
-    ρq4 = ρ_ref(z) * q4
+    ua =
+        k * sind(λp)^2 * sind(2 * ϕ) * cos(FT(π) * t / τ) +
+        2 * FT(π) * R / τ * cosd(ϕ)
+    ud =
+        ω_0 * R / b / p_top *
+        cosd(λp) *
+        cosd(ϕ)^2 *
+        cos(2 * FT(π) * t / τ) *
+        (-exp((p(z) - p_0) / b / p_top) + exp((p_top - p(z)) / b / p_top))
+    uu = ua + ud
 
-    return (
-        ρ = ρ_ref(z),
-        ρq1 = ρq1,
-        ρq2 = ρq2,
-        ρq3 = ρq3,
-        ρq4 = ρq4,
-        ρq1_td = ρq1,
-        ρq2_td = ρq2,
-        ρq3_td = ρq3,
-        ρq4_td = ρq4,
-    )
+    uv = k * sind(2 * λp) * cosd(ϕ) * cos(FT(π) * t / τ)
+
+    sp =
+        1 + exp((p_top - p_0) / b / p_top) - exp((p(z) - p_0) / b / p_top) -
+        exp((p_top - p(z)) / b / p_top)
+    ω = ω_0 * sind(λp) * cosd(ϕ) * cos(2 * FT(π) * t / τ) * sp
+    uw = -ω / ρ_ref(z) / grav
+
+    Geometry.Covariant123Vector(Geometry.UVWVector(uu, uv, uw), local_geometry)
 end
 
-y0 = Fields.FieldVector(
-    ρ = y0.ρ,
-    ρq1 = y0.ρq1,
-    ρq2 = y0.ρq2,
-    ρq3 = y0.ρq3,
-    ρq4 = y0.ρq4,
-    ρq1_td = y0.ρq1,
-    ρq2_td = y0.ρq2,
-    ρq3_td = y0.ρq3,
-    ρq4_td = y0.ρq4,
-)
+# "Tendency increment", which computes Y2 .+= dt * ∂Y/∂t(Y1, t)
+function increment!(Y2, Y1, cache, t, dt, beta)
+    (; cent_uvw, face_uvw, face_uv, face_w, limiter, Δₕq, fct_op, ρq_td) = cache
+    ρ1 = Y1.c.ρ
+    ρq1 = Y1.c.ρq
+    ρ2 = Y2.c.ρ
+    ρq2 = Y2.c.ρq
+
+    cent_local_geometry = Fields.local_geometry_field(cent_uvw)
+    face_local_geometry = Fields.local_geometry_field(face_uvw)
+    @. cent_uvw = local_flow(cent_local_geometry, t)
+    @. face_uvw = local_flow(face_local_geometry, t)
+    horz_axis = Ref(Geometry.Covariant12Axis())
+    vert_axis = Ref(Geometry.Covariant3Axis())
+    @. face_uv = Geometry.project(horz_axis, face_uvw)
+    @. face_w = Geometry.project(vert_axis, face_uvw)
+
+    # NOTE: With a limiter, the order of operations 1-5 below matters!
+
+    # 1) Horizontal transport:
+    @. ρ2 -= dt * hdiv(ρ1 * cent_uvw)
+    for n in 1:5 # TODO: update RecursiveApply/Operators to eliminate this loop
+        ρq1_n = ρq1.:($n)
+        ρq2_n = ρq2.:($n)
+        @. ρq2_n -= dt * hdiv(ρq1_n * cent_uvw)
+    end
 
-function rhs!(dydt, y, parameters, t, alpha, beta)
+    # 2) Limiter application:
+    if !isnothing(limiter)
+        Limiters.compute_bounds!(limiter, ρq1, ρ1)
```

Comment:
Interesting that when applying `compute_bounds` to a tuple of tracers I had encountered an error in the `compute_bounds` (#728 ), but now it seems to be resolved. Probably there were some changes in the data layout later on or some other changes. We could probably close #728 then

Reply:
Yeah; Simon's changes probably fixed it. I'll mark this as closing the issue.